src/ralph/api/auth.py

```diff
@@ -107,8 +106,50 @@ def get_stored_credentials(auth_file: Path) -> ServerUsersCredentials:
         raise AuthenticationError(msg.format(auth_file))
     return ServerUsersCredentials.parse_file(auth_file)
 
-
-def authenticated_user(credentials: HTTPBasicCredentials = Depends(security)):
+@lru_cache()
+def _scope_is_authorized(requested_scope, user_scopes: Tuple):
+    """Check if the requested scope can be accessed"""
+
+    # List of all expanded scopes
+    expanded_scopes = {
```

Comment:
Maybe we can use Pydantic to define these scopes

Reply:
Yes ! I will do this in a future iteration

Reply:
:sunglasses: